examples/trading/api/indexer/escrow-handler.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { SuiEvent } from '@mysten/sui.js/client';
+import { Escrow } from '@prisma/client';
+
+import { prisma } from '../db';
+
+type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>;
+
+type EscrowEvent = EscrowCreated | EscrowCancelled | EscrowSwapped;
+
+type EscrowCreated = {
+	sender: string;
+	recipient: string;
+	escrow_id: string;
+	key_id: string;
+};
+
+type EscrowSwapped = {
+	escrow_id: string;
+};
+
+type EscrowCancelled = {
+	escrow_id: string;
+};
+
+/// Handles all events emitted by the `lock` module.
+export const handleEscrowObjects = async (events: SuiEvent[]) => {
+	const updates: Record<
+		string,
+		Optional<Escrow, 'id' | 'keyId' | 'cancelled' | 'sender' | 'recipient' | 'swapped'>
+	> = {};
+
+	for (const event of events) {
+		const data = event.parsedJson as EscrowEvent;
+
+		if (!Object.hasOwn(updates, data.escrow_id)) {
+			updates[data.escrow_id] = {
+				objectId: data.escrow_id,
+			};
+		}
+
+		// Escrow cancellation case
+		if (event.type.endsWith('::EscrowCancelled')) {
+			const data = event.parsedJson as EscrowCancelled;
+			updates[data.escrow_id].cancelled = true;
+			continue;
+		}
+
+		// Escrow swap case
+		if (event.type.endsWith('::EscrowSwapped')) {
+			const data = event.parsedJson as EscrowSwapped;
+			updates[data.escrow_id].swapped = true;
+			continue;
+		}
+
+		const creationData = event.parsedJson as EscrowCreated;
+
+		// Handle creation event
+		updates[data.escrow_id].sender = creationData.sender;
+		updates[data.escrow_id].recipient = creationData.recipient;
+		updates[data.escrow_id].keyId = creationData.key_id;
+	}
+
+	/// SQLite does not support bulk insertion & conflict handling, so we have to insert 1 by 1.
+	//  Always use a single `bulkInsert` query with proper `onConflict` handling in production.
```

Comment:
Seeing as we are creating an example that people should use as a reference for production, we should be doing "the right thing" here (probably the set-up of `Display` in the previous PR deserved a similar comment...) what would it take to achieve that in this case?

Reply:
I wanted to avoid an external dependency of an actual prod DB, as sqlite is plug n play, and just creates a dev.db file. Someone could super easily configure this to work with Postgres, for instance, and just do a `bulkInsert +  on conflict ...` instead of writing to the db 1 by 1 like I do here. 

Prisma doesn't support bulk insert for sqlite to the extent of my research.

I can easily also change this to use Postgres, just worried on the extra steps needed here. Thoughts?

Reply:
Ah I see, that makes sense -- maybe add the clarification that you recommend using a different DB engine in production as well? It wasn't totally clear to me from reading the comment what the main reason was why we were doing this in the demo even though we shouldn't do it in production (even though I know you mention sqlite).